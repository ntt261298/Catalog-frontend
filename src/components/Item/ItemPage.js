import React from 'react';
import { connect } from 'react-redux';
import 'style/home.css';
import { PropTypes } from 'prop-types';
import ItemDetail from 'components/Item/ItemDetail';
import Header from 'components/Layout/Header';

export const ItemPage = props => (
  <div>
    <Header />
    <div className="item-main">
      <ItemDetail params={props.match.params} />
    </div>
  </div>
);

ItemPage.propTypes = {
  match: PropTypes.object.isRequired,
};

export default connect(null, null)(ItemPage);
