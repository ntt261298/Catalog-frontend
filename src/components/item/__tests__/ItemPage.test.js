import React from 'react';
import { shallow } from '../../../enzyme';
import { ItemPage } from '../ItemPage';

describe('src/components/layout/CategoryPage.js', () => {
  let wrapper;
  let props;

  const setup = () => {
    wrapper = shallow(
      <ItemPage {...props} />,
    );
  };

  beforeEach(() => {
    props = {
      match: {
        params: {
          categoryID: 1,
          userID: 2,
        },
      },
    };
  });

  it('should render item page', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });
});
