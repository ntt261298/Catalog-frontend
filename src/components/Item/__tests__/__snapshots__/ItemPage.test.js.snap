// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Item/ItemPage should render Item page 1`] = `
<div>
  <Connect(Component) />
  <div
    className="item-main"
  >
    <Connect(ItemDetail)
      params={
        Object {
          "categoryID": 1,
          "userID": 2,
        }
      }
    />
  </div>
</div>
`;
